lib/src/main/java/dev/openfeature/javasdk/OpenFeatureClient.java

```diff
@@ -1,135 +1,98 @@
 package dev.openfeature.javasdk;
 
-import com.google.common.collect.ImmutableList;
-import com.google.common.collect.ImmutableMap;
-import com.google.common.collect.Lists;
-import dev.openfeature.javasdk.exceptions.*;
-import lombok.Getter;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
+import java.util.*;
 
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.List;
-import java.util.Optional;
+import dev.openfeature.javasdk.exceptions.GeneralError;
+import dev.openfeature.javasdk.internal.ObjectUtils;
+import lombok.Getter;
+import org.slf4j.*;
 
-@SuppressWarnings("PMD.DataflowAnomalyAnalysis")
+@SuppressWarnings({"PMD.DataflowAnomalyAnalysis", "PMD.BeanMembersShouldSerialize", "unchecked", "rawtypes"})
 public class OpenFeatureClient implements Client {
-    private transient final OpenFeatureAPI openfeatureApi;
-    @Getter private final String name;
-    @Getter private final String version;
-    @Getter private final List<Hook> clientHooks;
     private static final Logger log = LoggerFactory.getLogger(OpenFeatureClient.class);
 
+    private final OpenFeatureAPI openfeatureApi;
+    @Getter
+    private final String name;
+    @Getter
+    private final String version;
+    @Getter
+    private final List<Hook> clientHooks;
+    private final HookSupport hookSupport;
+
     public OpenFeatureClient(OpenFeatureAPI openFeatureAPI, String name, String version) {
         this.openfeatureApi = openFeatureAPI;
         this.name = name;
         this.version = version;
         this.clientHooks = new ArrayList<>();
+        this.hookSupport = new HookSupport(log);
```

Comment:
Should this not generate it's own logger instance? My understanding of java logging (rusty!) is that generating it w/ the right class name makes for better logs.  Or are you thinking that b/c this is a private utility only for use in this context that sharing the logger is fine?

Reply:
Usually I don't pass loggers around. In this case, I wanted to make sure, that the logger stays the same, even after moving out the code, in case there is already a logging configuration, that I missed. As it doesn't seem to be that way, I will fetch a new logger in the `HookSupport` class, rather than passing it in.